tests/python/pants_test/test_base.py

```diff
@@ -753,3 +754,9 @@ def populate_target_dict(self, target_map):
       target_dict[key] = generated_target
 
     return target_dict
+
+
+def use_current_prerelease_semver(f):
+  return unittest.skipIf(not PANTS_SEMVER.is_prerelease,
+                         'Uses the current version as a deprecation version, which is only valid '
+                         'for prereleases.')(f)
```

Comment:
Rather than taking this approach of skipping if not pre-release, could we monkey patch in the relevant tests `pants.version.VERSION` to be whatever we want it to be? That way the unit tests are not dependent on some global state (the release version).

If we stick with this current approach, consider renaming this to `skip_if_pants_version_not_prerelease`. `use_current` suggests to me that you're somehow forcing it to use the the prerelease version.

Reply:
I've ended up using `@mock.patch('pants.base.deprecated.PANTS_SEMVER', ...)` to accomplish this!